test/e2e/mcd_test.go

```diff
@@ -967,3 +969,151 @@ func assertExpectedPerms(t *testing.T, cs *framework.ClientSet, node corev1.Node
 	actualPerms := strings.Split(strings.TrimSuffix(helpers.ExecCmdOnNode(t, cs, node, "chroot", "/rootfs", "stat", "--format=%U %G %a", path), "\n"), " ")
 	assert.Equal(t, expectedPerms, actualPerms, "expected %s to have perms %v, got: %v", path, expectedPerms, actualPerms)
 }
+
+// Tests that changes to the internal image registry pull secret has the
+// desired effect on both the ControllerConfig and the individual nodes
+// filesystems.
+func TestInternalImageRegistryPullSecret(t *testing.T) {
+	cs := framework.NewClientSet("")
+
+	hostnames := []string{
+		"registry1.hostname.com",
+		"registry2.hostname.com",
+		"registry3.hostname.com",
+	}
+
+	filename := "/etc/mco/internal-registry-pull-secret.json"
+	canonicalizedFilename := filepath.Join("/rootfs", filename)
+
+	// For each hostname, we do the following (in parallel to make the test
+	// faster and ensure that we don't run into any unforeseen edge-cases):
+	//
+	// 1. Create a new secret with the sanitized hostname as the secret name.
+	// 2. Append the secret name to the machine-os-puller service account.
+	// 3. Wait for the ControllerConfig to pick up the new hostname.
+	// 4. Wait for each of the nodes to pick up the new hostname.
+	// 5. Delete the secret.
+	// 6. Remove the secret name from the machine-os-puller service account.
+	// 7. Wait for the ControllerConfig to lose the hostname.
+	// 8. Wait for each of the nodes to lose the new hostname.
+	for _, imageRegistryHostname := range hostnames {
+		imageRegistryHostname := imageRegistryHostname
+		t.Run(imageRegistryHostname, func(t *testing.T) {
+			t.Parallel()
+
+			revertFunc := setupForInternalImageRegistryPullSecretTest(t, cs, imageRegistryHostname)
+			t.Cleanup(revertFunc)
+
+			t.Logf("Waiting for ControllerConfig to get hostname %s", imageRegistryHostname)
+
+			helpers.AssertControllerConfigReachesExpectedState(t, cs, func(cc *mcfgv1.ControllerConfig) bool {
+				return strings.Contains(string(cc.Spec.InternalRegistryPullSecret), imageRegistryHostname)
+			})
+
+			t.Logf("Waiting for all nodes to get hostname %s", imageRegistryHostname)
+
+			helpers.AssertAllNodesReachExpectedState(t, cs, func(node corev1.Node) bool {
+				contents := helpers.ExecCmdOnNode(t, cs, node, "cat", canonicalizedFilename)
```

Comment:
Heh, totally non blocking, but another handy place to use my get file content helper!

Reply:
If that lands before this does, I'll happily adopt it!